packages/@lwc/integration-karma/helpers/test-utils.js

```diff
@@ -328,9 +328,11 @@ window.TestUtils = (function (lwc, jasmine, beforeAll) {
 
     var customMatchers = {
         toLogErrorDev: consoleDevMatcherFactory('error'),
+        toLogError: consoleDevMatcherFactory('error', true),
         toLogWarningDev: consoleDevMatcherFactory('warn'),
         toThrowErrorDev: errorMatcherFactory(directErrorListener),
-        toThrowConnectedError: errorMatcherFactory(customElementConnectedErrorListener),
+        toThrowConnectedErrorDev: errorMatcherFactory(customElementConnectedErrorListener),
+        toThrowConnectedError: errorMatcherFactory(customElementConnectedErrorListener, true),
```

Comment:
Honestly we should probably audit every case where we throw an error in dev mode but not prod mode. It feels weird to have behavior that differs so significantly between dev and prod.

Reply:
Aren't most of those supposed to be extra checks to validate function inputs and provide more detailed information and not necessarily differences in behavior? Did you see otherwise?

Reply:
Agreed that this makes writing tests for both dev and prod modes more of a pain. At the same time, IMO, if part of the desired functionality is that we have more detailed checking and erroring in dev mode that don't show up in prod mode, we should have tests that verify that.

Sounds like you're asking whether all of it is needed right now; I agree we should audit that.

Reply:
Right now I'm just writing tests that confirm our existing behavior. I'm not changing the behavior.

> Aren't most of those supposed to be extra checks to validate function inputs and provide more detailed information and not necessarily differences in behavior? Did you see otherwise?

Yep, in plenty of places. Sometimes it varies between browsers, or sometimes between native and synthetic shadow, but other times we just straight-up have different behavior in dev vs prod mode. See: #3245